### PR TITLE
Provide a decent development environment

### PR DESCRIPTION
**Pull request by Drakulix**
This does a multitude of things:

- It fixes the evalData script to support the new login code
- It adds Docker/compose files to run a development environment that auto-reloads all changes source files upon change.
- It adds a small readme showing how to use it

Everybody may review please.

### CODE REVIEW

**Comment by Drakulix** [comment]
Wanted to do some smaller fixes and improvements, but my internet connection is bugging out and I cannot test it, so marking this as WIP. Not yet ready for merge. Still ready for testing and review.

File: docker-compose-dev.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        build:
+            context: ./
+            dockerfile: ./Dockerfile-dev-frontend
+        restart: "always"
+        depends_on:
+            - "flask"
+        volumes:
+            - ./web:/frontend
+        ports:
+            - "3000:3000"
```
Comment (henderkes) [inline docker-compose-dev.yml]: I like this much better than before

File: evalData/testdata_insertion.py [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
 import io
 import os
 import json
+import requests
 
 if __name__ == "__main__":
-    conn = http.client.HTTPConnection("localhost", 5000)
+    session = requests.Session()
+    data = {"email": "admin@knex.com", "password": "admin"}
+    session.post("http://localhost:3000/api/users/login", data=data)
```
Comment (henderkes) [inline evalData/testdata_insertion.py]: should these requests really go through the frontend?

**Comment by gitmirgut** [comment]
It works like a charm :tada: (for FE)

File: evalData/testdata_insertion.py (hunk 2 above)
Comment (Drakulix) [reply evalData/testdata_insertion.py]: I don't export the backend port and frontend does only proxy it, so why not.

File: docker-compose-dev.yml (hunk 1 above)
Comment (Drakulix) [reply docker-compose-dev.yml]: Better then what? The deployment file is still there (for deployment).

Comment (henderkes) [reply docker-compose-dev.yml]: Well it's hard to correctly mark the part I meant so it kind of landed in the nowhere. What I meant is also specifying the FE part in the compose yaml and mounting the volumes for instant code changes. We already had that in the BE once and it was very convenient. Also having a shared port to access the through the FE specifically, just better overall.

File: evalData/testdata_insertion.py (hunk 2 above)
Comment (henderkes) [reply evalData/testdata_insertion.py]: Oh, missed that part (BE no longer being exported on port 5000), perfectly makes sense then.

**Comment by Drakulix** [comment]
Ok. I think this is good now.

@DubbleClick @gitmirgut A last test and review before merge?

**Comment by henderkes** [comment]
I think travis doesn't agree

**Comment by Drakulix** [comment]
Okay now hopefully. @DubbleClick

**Review by henderkes — DISMISSED** [review]
looks good now

**Comment by henderkes** [comment]
@gitmirgut  @Rintel or anyone from frontend?

**Review by henderkes — DISMISSED** [review]
taking too long, aproving and merging this now

**Pull request merged**